src/FormBuilder/resources/js/form-builder/src/middleware/uniqueFieldNames.ts

```diff
@@ -0,0 +1,70 @@
+import {slugifyMeta} from "@givewp/form-builder/supports/field-settings/MetaKeyTextControl";
+import {flattenBlocks, getFieldNameValidator} from "@givewp/form-builder/hooks/useFieldNameValidator";
+import {FieldSettings} from "@givewp/form-builder/supports/field-settings/types";
+import {getBlockSupport} from "@wordpress/blocks";
+import {GiveWPSupports} from "@givewp/form-builder/supports/types";
+import normalizeFieldSettings from "@givewp/form-builder/supports/field-settings/normalizeFieldSettings";
+
```

Comment:
There's a lot of great stuff going on here, can we please add doc blocks, since tags, and types to this file